pkg/cover/report.go

```diff
@@ -123,6 +129,20 @@ func (rg *ReportGenerator) prepareFileMap(progs []Prog) (map[string]*file, error
 	if !matchedPC {
 		return nil, fmt.Errorf("coverage doesn't match any coverage callbacks")
 	}
+	// Check that all collected coverage points have matching coverage callbacks.
+	// This check is not performed for targets that explicitly list "gcc" as KernelCompiler,
+	// because code coverage instrumentation does not work correctly in GCC < 14. For Clang
+	// and/or targets that do not set the compiler (e.g. in tests) the check is still done.
+	// See https://github.com/google/syzkaller/issues/4447 for more information.
+	if (len(allProgPCs) != len(matchedProgPCs)) && (rg.target.KernelCompiler != "gcc") {
```

Comment:
@a-nogikh do our CI configs explicitly set KernelCompiler?

Reply:
Hmm, no, I think we do not set `rg.target.KernelCompiler` explicitly. 

When Linux kernel is built, it's only used as a back-up value:

https://github.com/google/syzkaller/blob/1c0ecc51adc9f44c7bd3f45c2aa1b62718d1236e/pkg/build/linux.go#L182

https://github.com/google/syzkaller/blob/1c0ecc51adc9f44c7bd3f45c2aa1b62718d1236e/pkg/build/linux.go#L193-L198

But we always set it in syz-ci, so in reality it's never used during builds:

https://github.com/google/syzkaller/blob/1c0ecc51adc9f44c7bd3f45c2aa1b62718d1236e/syz-ci/syz-ci.go#L425-L427

Reply:
To `sys/targets` the value is passed via an env variable (quite hacky)

https://github.com/google/syzkaller/blob/1c0ecc51adc9f44c7bd3f45c2aa1b62718d1236e/sys/targets/targets.go#L1005

But afaik we do not set it anywhere other than in some presubmit tests.

Reply:
I changed the code to read ELF's .comment section for the compiler version.
If any of the binaries is built with GCC, we skip the check.